src/lib.rs

```diff
@@ -345,12 +347,22 @@ pub mod pallet {
 
 			let current_initialized_rewards = InitializedRewardAmount::<T>::get();
 
-			// This ensures that the rewards match the pot
+			// There can't be more than 1 unit (10^-18) dust per reward so
 			ensure!(
-				current_initialized_rewards == Self::pot(),
+				current_initialized_rewards > Self::pot() - TotalContributors::<T>::get().into(),
 				Error::<T>::RewardsDoNotMatchFund
 			);
 
+			// Anything that does not match the pot should go to DustHandler
+			if Self::pot() > current_initialized_rewards {
+				T::RewardCurrency::transfer(
+					&PALLET_ID.into_account(),
+					&T::DustHandler::get(),
+					Self::pot().saturating_sub(current_initialized_rewards),
+					AllowDeath,
+				)?;
```

Comment:
```suggestion
      let pot = Self::pot();
			if pot > current_initialized_rewards {
				T::RewardCurrency::transfer(
					&PALLET_ID.into_account(),
					&T::DustHandler::get(),
					pot.saturating_sub(current_initialized_rewards),
					AllowDeath,
				)?;
```
only get balance of the pot account once instead of twice